### PR TITLE
[RLlib; RELEASE BLOCKER] Fix Policy server/client (currently broken and not caught by tests!)

### PR DESCRIPTION
**Pull request by sven1977**
Fix Policy server/client (currently broken and not caught by tests!)

* Fixes broken local inference RolloutWorker creation on a) client side (mode "local") or b) server side (mode "remote").
* Enhances test cases to test for learning success. The current tests don't even notice if one or more client processes crash due to a failed connection/failed server.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/env/policy_client.py [hunk 1]
```diff
@@ -373,26 +373,25 @@ def _create_embedded_rollout_worker(kwargs, send_fn):
...
-    del kwargs["input_creator"]
-
-    # Since the server also acts as an output writer, we might have to reset
-    # the output config to the default, i.e. "output": None, otherwise a
-    # local rollout worker might write to an unknown output directory
-    del kwargs["output_creator"]
+    kwargs["config"] = kwargs["config"].copy(copy_frozen=False)
+    config = kwargs["config"]
+    config.output = None
+    config.input_ = "sampler"
```
Comment (avnishn) [inline rllib/env/policy_client.py:379]: lgtm. Was thinking that users might need access to other input types but I guess why would anyone use policy_client if they weren't doing environment sampling

**Review by avnishn — APPROVED** [review]

File: rllib/env/policy_server_input.py [hunk 2]
```diff
@@ -64,7 +65,7 @@ class PolicyServerInput(ThreadingMixIn, HTTPServer, InputReader):
     """
 
     @PublicAPI
-    def __init__(self, ioctx, address, port, idle_timeout=3.0):
+    def __init__(self, ioctx, address, port, idle_timeout=3.0, use_json=False):
```
Comment (avnishn) [inline rllib/env/policy_server_input.py:68]: you've added this as an option, but is this used at any point now in the tests/examples?

**Comment by avnishn** [comment]
you mention that "The current tests don't even notice if one or more client processes crash due to a failed connection/failed server." 

but I'm not sure how the tests as they are rebuffed handle this issue. 
Can you comment on the lines where this is handled in your pr?

File: rllib/env/policy_client.py (hunk 1 above)
Comment (sven1977) [reply rllib/env/policy_client.py:379]: Yeah, that, too. But also note this is only for the extra RolloutWorker that we create on either the client (inference-mode=local) or on the server (inside the PolicyServerInput object!) for inference mode=remote.
The whole design is completely flawed imo, but we'll have to fix this separately, it's beyond the scope of this quick fix PR. This PR does NOT touch the original (bad) design.

File: rllib/env/policy_server_input.py (hunk 2 above)
Comment (sven1977) [reply rllib/env/policy_server_input.py:68]: Good catch, will remove.
Leftover from my other work, which made me discover this bug in the first place :)

**Review by kouroshHakha — COMMENTED** [review]
left a couple of questions

**Review by kouroshHakha — COMMENTED** [review]
left a couple of questions

**Review by kouroshHakha — APPROVED** [review]
approved.

**Pull request merged**